File: preschool-management-backend/src/main/java/com/vusystem/preschool_management_backend/modules/mobile/service/MedicationService.java
package com.vusystem.preschool_management_backend.modules.mobile.service;

import com.vusystem.preschool_management_backend.modules.mobile.dto.request.MedicationCreateRequest;
import com.vusystem.preschool_management_backend.modules.mobile.dto.response.MedicationResponse;

import java.time.LocalDate;
import java.util.List;

public interface MedicationService {
    MedicationResponse createRequest(MedicationCreateRequest request);

    List<MedicationResponse> getParentRequests(Long childId);

    // phân tách api cho giáo viên, cần map thêm cảnh báo dị ứng từ profile học sinh
    List<MedicationResponse> getClassRequests(Long classId, LocalDate date);

    void markAsCompleted(Long id);
}

File: mobile/src/services/medication.service.ts

import axiosInstance from '../config/api.client';

export interface MedicationCreateRequest {
  childId: number;
  medicationName: string;
  dosage: string;
  startDate: string;
  endDate: string;
  notes: string;
}

export interface MedicationResponse {
  id: number;
  childId: number;
  childFullName: string;
  medicationName: string;
  dosage: string;
  startDate: string;
  endDate: string;
  notes: string;
  status: 'PENDING' | 'COMPLETED';
  allergies?: string[];
}

export const medicationService = {
  createRequest: async (data: MedicationCreateRequest): Promise<MedicationResponse> => {
    const response = await axiosInstance.post<{ status: number; message: string; data: MedicationResponse }>(
      '/mobile/medications',
      data
    );
    return response.data.data;
  },

  getParentRequests: async (childId: number): Promise<MedicationResponse[]> => {
    const response = await axiosInstance.get<{ status: number; message: string; data: MedicationResponse[] }>(
      `/mobile/medications/children/${childId}`
    );
    return response.data.data;
  },

  getClassRequests: async (classId: number, date: string): Promise<MedicationResponse[]> => {
    const response = await axiosInstance.get<{ status: number; message: string; data: MedicationResponse[] }>(
      `/mobile/medications/classes/${classId}`,
      { params: { date } }
    );
    return response.data.data;
  },

  markAsCompleted: async (id: number): Promise<void> => {
    await axiosInstance.put(`/mobile/medications/${id}/complete`);
  }
};
